src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisContextInfo.cs

```diff
@@ -29,41 +31,48 @@ internal struct AnalysisContextInfo
         }
 
         public AnalysisContextInfo(Compilation compilation, ISymbol symbol) :
-            this(compilation: compilation, operation: null, symbol: symbol, tree: null, node: null)
+            this(compilation: compilation, operation: null, symbol: symbol, tree: null, node: null, nonSourceFile: null)
         {
         }
 
         public AnalysisContextInfo(Compilation compilation, SyntaxTree tree) :
-            this(compilation: compilation, operation: null, symbol: null, tree: tree, node: null)
+            this(compilation: compilation, operation: null, symbol: null, tree: tree, node: null, nonSourceFile: null)
+        {
+        }
+
+        public AnalysisContextInfo(Compilation compilation, AdditionalText nonSourceFile) :
+            this(compilation: compilation, operation: null, symbol: null, tree: null, node: null, nonSourceFile)
         {
         }
 
         public AnalysisContextInfo(Compilation compilation, SyntaxNode node) :
-            this(compilation: compilation, operation: null, symbol: null, tree: node.SyntaxTree, node: node)
+            this(compilation: compilation, operation: null, symbol: null, tree: node.SyntaxTree, node, nonSourceFile: null)
         {
         }
 
         public AnalysisContextInfo(Compilation compilation, IOperation operation) :
-            this(compilation: compilation, operation: operation, symbol: null, tree: operation.Syntax.SyntaxTree, node: operation.Syntax)
+            this(compilation: compilation, operation: operation, symbol: null, tree: operation.Syntax.SyntaxTree, node: operation.Syntax, nonSourceFile: null)
         {
         }
 
         public AnalysisContextInfo(Compilation compilation, ISymbol symbol, SyntaxNode node) :
-            this(compilation: compilation, operation: null, symbol: symbol, tree: node.SyntaxTree, node: node)
+            this(compilation: compilation, operation: null, symbol: symbol, tree: node.SyntaxTree, node, nonSourceFile: null)
         {
         }
 
         public AnalysisContextInfo(
-            Compilation compilation,
-            IOperation operation,
-            ISymbol symbol,
-            SyntaxTree tree,
-            SyntaxNode node)
+            Compilation? compilation,
+            IOperation? operation,
+            ISymbol? symbol,
+            SyntaxTree? tree,
+            SyntaxNode? node,
+            AdditionalText? nonSourceFile)
```

Comment:
Consider using optional parameters to make the constructor chain of this type simpler.

Reply:
This constructor was only meant for private use, so internal constructors only need to specify non-null arguments. I have changed this to be private now.